presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarImplementation.java

```diff
@@ -473,6 +478,23 @@ else if (parameterType.isPrimitive() && !hasNullFlag) {
             }
         }
 
+        private void checkTypeParameters(TypeSignature signature, Method method, ImmutableSet<String> typeParameterNames)
```

Comment:
Rename `signature` to `typeSignature`. In the context of this class, signature is more often used to refer to method signatures.